### PR TITLE
chore: add debug command to resync resources - WPB-14794

### PR DESCRIPTION
**Pull request by johnxnguyen**
### Issue

This PR:

- adds a developer tool to delete domains in the database, useful for debugging federation issues
- adds a developer tool to trigger resync of resources
- add a debug command to trigger resync of resources (useful for users who need to repair their database)
- fix trigger of resync resources (notifying new requests available)
- 
---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DebugActions.swift [hunk 1]
```diff
@@ -242,8 +242,10 @@ enum DebugActions {
             switch command {
             case .repairInvalidAccessRoles:
                 DebugActions.updateInvalidAccessRoles()
-            }
 
+            case .resyncResources:
+                DebugActions.triggerResyncResources()
+            }
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/DebugActions.swift:248]: can we expose slow sync? I guess it's not strictly needed

File: wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DebugActions/DeveloperDebugActionsViewModel.swift [hunk 2]
```diff
@@ -159,4 +167,37 @@ final class DeveloperDebugActionsViewModel: ObservableObject {
...
+
+                for conversation in conversations where conversation.conversationType.isOne(of: .oneOnOne, .group) {
+                    conversation.domain = nil
+                }
+
+                try syncContext.save()
+                print("successfully deleted domains")
+
+            } catch {
+                print("failed to delete domains: \(error.localizedDescription)")
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DebugActions/DeveloperDebugActionsViewModel.swift]: can we use a WireLogger might be useful to check on DD

Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/Developer/DeveloperTools/DebugActions/DeveloperDebugActionsViewModel.swift]: done

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DebugActions.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/DebugActions.swift:248]: if it's not needed maybe we skip it for now and revisit in the next version?

**Review by netbe — APPROVED** [review]
let's ship it

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**